### ISSUE TITLE
:lab: always links to lab.uberspace.de

### ISSUE THREAD

**Issue by luto** [Uberspace/lab#331]
`:lab:` always links to lab.uberspace.de, not to the current domain. This is a problem in development setups and may be a problem in branch-domains, although I have not tested that.

**Comment by FriedrichFroebel** [comment]
This is something I already thought about when introducing the lab directives, but settled with the current approach (using lab.uberspace.de as explicit target) to keep it as simple as possible.

One way to circumvent this could be to do this like the Pelican static site generator which uses one configuration file for development and another one for publishing - which in fact just imports everything from the development one and overwrites some of the configuration values afterwards. As the `extlinks` plugin is configured with a dictionary, replacing these values should not be the problem. This approach would need changes in the Makefiles (additional tasks) and probably is feasible with a known number of static setups only (for example for production and development purposes with a specific localhost port, but not for Netlify deployment as this includes the number of the PR inside the URL).

**Comment by luto** [comment]
What about using domain-less links like `<a href="/guide_mailman-3.html"`?

**Comment by FriedrichFroebel** [comment]
You are right, I did not think about this. Seems like the following should be used instead (did a quick test only):

```python
extlinks = {
    'manual': (
        'https://manual.uberspace.de/%s.html',
        ''
    ),
    'manual_anchor': (
        'https://manual.uberspace.de/%s',
        ''
    ),
    'lab': (
        '%s.html',
        ''
    ),
    'lab_anchor': (
        '%s',
        ''
    ),
}
```

As I just stumbled over the `requirements.txt` using a Sphinx version from Git, but mentioning version 1.7.3: This version has been released nearly a year ago, the current one is 1.8.4 (see [Release history](https://pypi.org/project/Sphinx/#history)).

**Comment by FriedrichFroebel** [comment]
Another note: I just noticed that the only usage of the `:lab_anchor:` directive inside the Miniflux guide is missing the HTML extension and should read

```rst
:lab_anchor:`Guide <guide_postgresql.html#database-and-user-management>`
```

**Comment by ebroda** [comment]
@luto isn't this closed by the merged PR?

**Comment by luto** [comment]
Because the PR description did not say "closes #331".
